src/priority-queue.ts

```diff
@@ -0,0 +1,257 @@
+import {ArmorCollection} from './collection';
+import {ArmorCollectionSelector} from './selector';
+import ArmorPriorityQueueComparator from './priority-queue-comparator';
+import ArmorPriorityQueueNodeChildren from './priority-queue-children';
+import ArmorPriorityQueueOptions from './priority-queue-options';
+
+export default class ArmorPriorityQueue<T> implements ArmorCollection<T> {
+	public elements: T[];
+	public readonly comparator: ArmorPriorityQueueComparator<T>;
+
+	constructor(comparator: ArmorPriorityQueueComparator<T>, options?: ArmorPriorityQueueOptions<T>) {
+		if (typeof comparator !== 'function') {
+			throw new Error('Must have a comparator function for priority queue to operate properly');
+		}
+
+		this.elements = [];
+		this.comparator = comparator;
+
+		if (options) {
+			this.parseOptions(options);
+		}
+	}
+
+	public parseOptions(options: ArmorPriorityQueueOptions<T>): void {
+		if (options.elements !== undefined) {
+			if (!Array.isArray(options.elements)) {
```

Comment:
- [ ] Replace the contents of the isArray conditional with call to parse (defined in comments above).

You can gut the entire contents of everything inside the if elements is an array check, and replace it with:
```typescript
const elements = this.parse(options.elements);
if (elements) {
   this.elements = elements;
}
```